packages/design-system/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,411 +0,0 @@
-      document.removeEventListener('keydown', handleEscapeKey);
-    };
-  }, [handleClickOutside, handleEscapeKey]);
-
-  useEffect(() => {
-    if (active) {
-      props.onOpen && props.onOpen();
-    } else {
-      props.onClose && props.onClose();
-
-      // if tooltip goes from active to inactive and is the dialog version, focus the trigger
-      if (prevActiveStateVar && (props.dialog || isMobile) && props.showCloseButton) {
-        if (triggerElement && triggerElement.current) {
-          triggerElement.current.focus();
-        }
-      }
-    }
-  }, [active]);
-
-  useLayoutEffect(() => {
-    if (popper.current) {
-      popper.current.setOptions(props);
-      popper.current.forceUpdate();
-    }
-  });
-
-  const renderTrigger = (props: TooltipProps): React.ReactElement => {
-    const {
-      activeClassName,
-      ariaLabel,
-      children,
-      className,
-      component,
-      dialog,
-      id,
-      offset,
-      onClose,
-      onOpen,
-      inversed,
-      interactiveBorder,
-      placement,
-      maxWidth,
-      title,
-      transitionDuration,
-      zIndex,
-      showCloseButton,
-      closeButtonLabel,
-      contentHeading,
-      ...others
-    } = props;
-
-    const TriggerComponent = component;
-    const triggerClasses = classNames('ds-c-tooltip__trigger', className, {
-      [activeClassName]: activeClassName && active,
-      'ds-c-tooltip__trigger--inverse': inversed,
-    });
-    const linkTriggerOverrides = {
-      tabIndex: TriggerComponent === 'a' ? 0 : undefined,
-    };
-    const eventHandlers = dialog
-      ? {
-          onTouchStart: () => handleTouch(),
-          onClick: () => {
-            if (!isMobile) {
-              setActive(!active);
-            }
-          },
-        }
-      : {
-          onTouchStart: () => handleTouch(),
-          onClick: () => {
-            if (!isMobile) {
-              setActive(!active);
-            }
-          },
-          onFocus: () => setActive(true),
-          onBlur: (event) => handleBlur(event),
-        };
-
-    return (
-      <TriggerComponent
-        type={TriggerComponent === 'button' ? 'button' : undefined}
-        aria-label={ariaLabel || undefined}
-        aria-describedby={dialog ? undefined : contentId}
-        className={triggerClasses}
-        ref={setTriggerElement}
-        {...others}
-        {...linkTriggerOverrides}
-        {...eventHandlers}
-      >
-        {children}
-      </TriggerComponent>
-    );
-  };
-
-  const renderContent = (props: TooltipProps): React.ReactElement => {
-    const {
-      closeButtonLabel,
-      dialog,
-      contentHeading,
-      inversed,
-      interactiveBorder,
-      placement,
-      maxWidth,
-      showCloseButton,
-      title,
-      transitionDuration,
-      zIndex,
-    } = props;
-
-    const tooltipStyle = { maxWidth, zIndex };
-    const interactiveBorderStyle: React.CSSProperties = {
```

Comment:
I imagine this is to overcome differences in styles between themes, but it's probably unnecessary. I agree with your observation that a lot of these old props can be cleaned out.